File: src/main/kotlin/net/dankito/k8s/domain/model/ResourceItem.kt
package net.dankito.k8s.domain.model

import java.time.Instant

open class ResourceItem(
    val name: String,
    val namespace: String? = null,
    val creationTimestamp: Instant,
    val highlightedItemSpecificValues: List<ItemValue> = emptyList(),
    val secondaryItemSpecificValues: List<ItemValue> = emptyList()
) {

    /**
     * The id attribute in HTML may only contain letters, digits, '_' and '-'
     */
    val htmlSafeId by lazy { "${namespace}__$name" }

    override fun toString() = "${namespace?.let { "$it " }}$name"

}

File: frontend/src/ts/model/ResourceItem.ts
import {ItemValue} from "./ItemValue"
import {ContainerStatus} from "./ContainerStatus"

export class ResourceItem {

  constructor(
    readonly name: string, readonly namespace?: string,
    readonly creationTimestamp?: string,
    readonly highlightedItemSpecificValues: ItemValue[] = [],
    readonly secondaryItemSpecificValues: ItemValue[] = [],
    readonly htmlSafeId: string = "", // TODO: required?

    // Pod only fields
    readonly status?: string,
    readonly podIP?: string,
    readonly container: ContainerStatus[] = [],
  ) { }

}